src/goto-instrument/dump_c.cpp

```diff
@@ -1380,8 +1379,7 @@ void dump_ct::cleanup_expr(exprt &expr)
             // add a typecast for NULL or 0
             if(
               argument.id() == ID_constant &&
```

Comment:
Could this also be refactored to `argument.is_constant()`? Or is this not possible?

Reply:
Yes, which should likely be cleaned up across the code base. May I defer this to a PR of its own?

Reply:
Yes, no problem.

Reply:
Now done in https://github.com/diffblue/cbmc/pull/6542.